Support slice ops with default start

pssrawat opened the pull request:
Summary:
Since a recent change we see None arg in the slice op

This is causing failure during XNNPack delegation, since XNNPack slice pass is trying to compare start_idx 'None' to 0.
This diff fixes that.

Differential Revision: D72503552

mcr229 reviewed (approved):
Thanks for the contribution :)

pssrawat committed:
Support slice ops with default start (#9923)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/9923

Since D71962884, we see the following slice ops in ASR encoder:
 {F1976830836}
This is causing failure during XNNPack delegation, since XNNPack slice pass is trying to compare start_idx 'None' to 0.
This diff fixes that.

Reviewed By: mcr229

Differential Revision: D72503552
